core/src/test/java/org/testcontainers/junit/DockerfileTest.java

```diff
@@ -0,0 +1,74 @@
+package org.testcontainers.junit;
+
+import com.github.dockerjava.api.command.BuildImageCmd;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.pass;
+
+public class DockerfileTest {
+
+    @Test
+    public void simpleDockerfileWorks() {
+        ImageFromDockerfile image = new ImageFromDockerfile()
+                .withFileFromString("folder/someFile.txt", "hello")
+                .withFileFromClasspath("test.txt", "mappable-resource/test-resource.txt")
+                .withFileFromClasspath("Dockerfile", "mappable-dockerfile/Dockerfile");
+
+        verifyImage(image);
+    }
+
+    @Test
+    public void customizableImage() {
+        ImageFromDockerfile image = new ImageFromDockerfile() {
+            @Override
+            protected void configure(BuildImageCmd buildImageCmd) {
+                super.configure(buildImageCmd);
+
+                List<String> dockerfile = Arrays.asList(
+                        "FROM alpine:3.2",
+                        "RUN sleep 1",
+                        "CMD ps"
+                );
+                withFileFromString("Dockerfile", String.join("\n", dockerfile));
+                buildImageCmd.withNoCache();
+            }
+        };
+
+        verifyImage(image);
+    }
+
+    @Test
+    public void dockerfileBuilderWorks() {
+        ImageFromDockerfile image = new ImageFromDockerfile()
+                .withFileFromClasspath("test.txt", "mappable-resource/test-resource.txt")
+                .withFileFromString("folder/someFile.txt", "hello")
+                .withDockerfileFromBuilder(builder -> builder
+                        .from("alpine:3.2")
```

Comment:
I think I like this approach to builders; it seems to get over some of the slight mental overheads of using builders for first-time users.

Perhaps for the long term (v2.0!) this pattern would be an improvement for the API in general, as @ostacey [suggests](https://github.com/testcontainers/testcontainers-java/issues/107#issuecomment-203136258).